integration-tests/tests/cases/mod.rs

```diff
@@ -217,3 +220,66 @@ async fn test_batch_duplicate_signature() -> anyhow::Result<()> {
     actions::batch_duplicate_signature_production(&nodes).await?;
     Ok(())
 }
+
+#[test(tokio::test)]
+async fn test_resharing_offline_participant_recovers() -> anyhow::Result<()> {
+    // have a short timeout for the resharing to complete in tests
+    set_resharing_running_timeout(Duration::from_secs(20));
+
+    let mut nodes = cluster::spawn().disable_prestockpile().await?;
+    nodes.wait().signable().await?;
+    let initial_state = nodes.expect_running().await?;
+    let initial_epoch = initial_state.epoch;
+
+    // Shutdown the node that will be offline during the resharing initially.
+    // This node will not appear in our local cluster list but still be a part of the
+    // contract participants. Meaning they're not kicked, just offline for resharing.
+    // They will have to restart later for resharing to complete.
+    let offline_account = nodes.account_id(0).clone();
+    let offline_config = nodes.kill_node(&offline_account).await;
+
+    // Start a new node that will be added during the resharing.
+    let new_account = nodes.start(None).await?;
+
+    // Voting in the new participant with threshold number of online participants
+    let participant_accounts = nodes.participant_accounts().await?;
+    let voters = participant_accounts
+        .into_iter()
+        .filter(|account| account.id() != &offline_account)
+        .take(nodes.cfg.threshold)
+        .collect::<Vec<_>>();
+    utils::vote_join(&voters, nodes.contract().id(), new_account.id()).await?;
+
+    // Wait for all online nodes to move to resharing state.
+    nodes.wait().nodes_resharing().await?;
+
+    // Now we should wait to see that we are still in the resharing state even after
+    // a long time, since one participant is offline and cannot complete the resharing.
+    tokio::time::sleep(Duration::from_secs(30)).await;
+    assert!(matches!(
+        nodes.contract_state().await?,
+        mpc_contract::ProtocolContractState::Resharing(_)
```

Comment:
How much effort would it take to fix these two issues? It is still possible to be "stuck" in the resharing state.
- https://github.com/sig-net/mpc/issues/272
- https://github.com/sig-net/mpc/issues/144

Reply:
added cancelling resharing here: https://github.com/sig-net/mpc/pull/570

Reply:
If we are kicking someone because of their misbehavior, the voting process will take several hours in the best-case scenario. The resharing protocol is relatively fast, so it does not matter much (5h vs 5h+1m).

The main reason we wanted parallel signing and resharing was possible downtime. Especially, downtime caused by failed resharing.

I'm more worried about https://github.com/sig-net/mpc/issues/11. Especially in a case where we want to kick the offline node.